src/hooks/useWaitForNavigation.ts

```diff
@@ -8,21 +8,18 @@ type UseWaitForNavigation = (navigate: () => void) => () => Promise<void>;
  * Only use when navigating by react-navigation
  */
 export default function useWaitForNavigation(): UseWaitForNavigation {
-    const navigation = useNavigation();
     const resolvePromises = useRef<Array<() => void>>([]);
 
-    useEffect(() => {
-        const unsubscribeBlur = navigation.addListener('blur', () => {
-            resolvePromises.current.forEach((resolve) => {
-                resolve();
-            });
-            resolvePromises.current = [];
-        });
-
-        return () => {
-            unsubscribeBlur();
-        };
-    }, [navigation]);
+    useFocusEffect(
```

Comment:
@adamgrzybowski @WojtekBoman could you also please review this change in case you could see any issue with it in future?

Reply:
I don't see anything weird about these changes. 

One thing I wonder is why `useFocusEffect` works and `addEventListnere('blur', ()=> {...})` doesn't while `useFocusEffect` uses the `blur` event underneath 🤔 

https://github.com/react-navigation/react-navigation/blob/2a745c8c598f95fcec5bbf5442045478d4046663/packages/core/src/useFocusEffect.tsx#L95-L101

Reply:
@adamgrzybowski because `useFocusEffect` runs cleanup on unmount (in native-stack we can't get `blur` event n that code)